src/containers/UsernameTranslation.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from "react-router-dom";
+
+/*
+At the moment works ONLY with apiPath='seacat_auth'
+- userId (required, array) - user ids to be translated
+- apiPath - default (and only working so far) 'seacat_auth'
+- cleanupTime - default set to 24 hours
+
+Usage:
+
+import { UsernameTranslation } from 'asab-webui';
+...
+<UsernameTranslation app={props.app} id={['user1-id', 'user2-id', 'user3-id']} apiPath='seacat_auth' cleanupTime={1000 * 60 * 60 * 24 * 14}/>
+
+*/
+
+export function UsernameTranslation(props) {
+
+	const { t } = useTranslation();
+
+	const apiPath = props.apiPath ?? 'seacat_auth';
+	let API = props.app.axiosCreate(apiPath);
+
+	const userIdsArray = Array.isArray(props.userId) ? props.userId : [props.userId] ;
+
+	const cleanupTime = props.cleanupTime ?? 1000 * 60 * 60 * 24; // 24 hrs
+
+	const [usernames, setUsernames] = useState([]);
+
+	// asks the server for usernames, saves them to local storage and sets usernames to render
+	const retrieveUserNames = async () => {
+		try {
+			let response = await API.put(`usernames`, userIdsArray);
+			if (response.data.result !== "OK") {
+				throw new Error(t("ASABUsernameTranslation|Something went wrong, failed to fetch assigned credentials"));
+			}
+			if (Object.keys(response.data.data).length === 0) {
+				throw new Error('..no data');
+			}
+			const usernamesToLS = setUsernamesToLS(response.data.data, userIdsArray, cleanupTime);
+			setUsernames(usernamesToLS);
+		} catch (e) {
+			props.app.addAlert("warning", t("ASABUsernameTranslation|Something went wrong, failed to fetch assigned credentials"));
+		}
+	}
+
+	// compares array of IDs with data in localstorage
+	const matchUserIds = (usernameIDs) => {
```

Comment:
There is no `usernameIDs` or `UserIds` ... there are `credentials_id` and `username`s.

Reply:
i thought `credentials_id` would stand for `'mongodb:ext:1234567890'` and `username` would be `'Ringo'`.
`usernameIDs` on line 50 expects to be an array of strings

Reply:
* `credentials_id` == 'mongodb:ext:1234567890' ... correct
* `username` == "Ringo" ... correct
* `credentials_ids` should be the array of `credentials_id`